spec.md

```diff
@@ -245,6 +243,32 @@ querying of the Service Broker's catalog, the Platform might not have an
 end-user with which to associate the request, therefore in those cases the
 originating identity header would not be included in those messages.
 
+## Service Broker Errors
+
+When a request to a Service Broker fails, the Service Broker MUST return an
+appropriate HTTP response code. If an appropriate response code is not defined
+within this specification, Service Broker authors SHOULD chose a code that
+accurately reflects the nature of the failure.
+
+For error responses, the following fields are valid:
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| error | string | A single word uniquely identifying the error condition. |
+| description | string | A meaningful error message explaining why the request failed. | 
+
+### Error Codes
+
+There are failure scenarios described throughout this specification for which
+the `error` field MUST contain a specific string. Service Broker authors MUST
+use these error codes for the specified failure scenarios. For other scenarios,
+Service Broker authors SHOULD choose a suitable one-word error code.
+
+| Error | Reason | Expected Action |
+| --- | --- | --- |
+| AsyncRequired | This request requires client support for asynchronous service operations. | The query parameter `accepts_incomplete=true` MUST be included the request. |
+| ConcurrencyError | The Service Broker does not support concurrent requests that mutate the same resource. | Clients SHOULD wait until pending requests have completed for the specified resources. |
```

Comment:
I wonder if this should be a MUST instead. I would think that for the same resource the broker wouldn't change whether it supports concurrency or not - unless someone can think of a case where the use of this error would vary based on which fields in the resource are being touched.

Reply:
I remember saying it's difficult to specify MUSTs for a client. Hmm.

Reply:
We use MUST in the previous one, so we wouldn't be inconsistent.

I would need to think more about whether a MUST for a client is an issue or not in general, but I think in this case it makes a lot of sense. This table is basically telling the client why it got an error, so saying that it "MUST do xxx" to avoid the error seems appropriate to me. Saying it "MAY do xxx" could be interpreted as "there might be other things it could do", or "try it again with the same data and next time maybe it'll work" - neither of which is what we're trying to say. I think the "expected action" column saying "you MUST do this to avoid the error" is exactly what people are looking for - they want to know what to do to fix it.

Reply:
@mattmcneeney what about this one?  MUST?

Reply:
Changed to MUST